File: solutions/c/raindrops/1/raindrops.c
#include "raindrops.h"
#include <string.h>
#include <stdio.h>

void convert(char result[], int number) {
    result[0] = '\0';

    if (number % 3 == 0) {
        strcat(result, "Pling");    
    } 
    if (number % 5 == 0) { 
        strcat(result, "Plang");
    }
    if (number % 7 == 0) {
        strcat(result, "Plong");    
    } 
    if (result[0] == '\0') {
        sprintf(result, "%d", number);
    }
}

File: solutions/elixir/raindrops/1/lib/raindrops.ex
defmodule Raindrops do

  def convert(number) do
    raindrops = [
       &if rem(&1, 3) == 0 do "Pling" else "" end,
       &if rem(&1, 5) == 0 do "Plang" else "" end,
       &if rem(&1 ,7) == 0 do "Plong" else "" end
    ]
    result = raindrops |> Enum.map_join("", & &1.(number))
    if result === "" do Integer.to_string(number) else result end
  end
  
end
